modules/controllers/jobs-controller.ts

```diff
@@ -31,27 +76,53 @@ export function JobsController(tracer?: any) {
             } = config[chain];
 
             // Guard against unconfigured chains
-            if (!balancerV3) {
+            if (!balancerV3 || !balancerPoolsV3 || !vaultAddress) {
                 throw new Error(`Chain not configured: ${chain}`);
             }
 
-            const vaultSubgraphClient = new BalancerVaultSubgraphSource(balancerV3);
-            const poolSubgraphClient = getPoolsSubgraphClient(balancerPoolsV3!);
+            const pools = await prisma.prismaPool.findMany();
+            const ids = pools.map((pool) => pool.id);
+            const client = getV3JoinedSubgraphClient(balancerV3, balancerPoolsV3);
+            const newPools = await client.getAllInitializedPools({ id_not_in: ids });
+
             const viemClient = getViemClient(chain);
             const latestBlock = await viemClient.getBlockNumber();
 
-            // TODO: add syncing v2 pools as well by splitting the poolService into separate
-            // actions with extracted configuration
+            await upsertPools(newPools, viemClient, vaultAddress, chain, latestBlock);
+        },
+        /**
+         * Takes all the pools from subgraph, enriches with onchain data and upserts them to the database
+         *
+         * @param chainId
+         */
+        async reloadPools(chainId: string) {
```

Comment:
would move this to the pool controller, as it is not triggered by a job

Reply:
i think we start to see a pattern emerging with organising things around models. looks like jobs becomes pretty much pools at the moment. how about forgetting about jobs specific controller, and use "model/domain" naming instead?

Reply:
I see what you mean. We can evolve it later